compose-example/docker-compose.yaml

```diff
@@ -0,0 +1,92 @@
+version: "3"
+
+services:
+  # The JupyterHub service configuration {
+  hub:
+    image: jupyterhub-traefik-proxy:example
+    build: .
+    container_name: jupyterhub
+
+    # Start traefik first
+    depends_on:
+      - traefik
+
+    volumes:
+      # Jupyterhub configuration file
+      - ./jupyterhub_config.py:/srv/jupyterhub/jupyterhub_config.py:ro
+
+      # Shared volume for the file provider's dynamic config
+      - traefik-dynamic-config:/var/run/traefik/
+
+      # jupyterhub's DockerSpawner needs read access to the docker socket.
```

Comment:
I'm surprised it only needs read access 🤷‍♂️

Reply:
Yeah, I don't think services should be modifying their own configuration file.

e.g. all config files in (at least my) `/etc` folder are only modifiable by root, but services are (normally / always?) run by a non-root user, so no service on my host PC can modify its own config file. Instead, services get write access to various subfolders in `/var` (e.g. `run`, `cache`, `db`, etc.).